spot-ml/SPARKCONF.md

```diff
@@ -120,34 +122,64 @@ and 50 GB. Driver maximum results should be something equal or bigger than 8 GB.
         SPK_DRIVER_MEM='30475m'
         SPK_DRIVER_MAX_RESULTS='8g'
         SPK_DRIVER_MEM_OVERHEAD='3047'
-        
- ![DriverMemory](https://raw.githubusercontent.com/Open-Network-Insight/oni-docs/master/images/Driver%20Memory.png)
+
  
  _Representation of memory allocation in driver node._
  
- 
  For more information about Spark properties click [here](http://spark.apache.org/docs/latest/configuration.html).
  
-###Known Spark error messages running oni-ml
+ ##### Spark autoBroadcastJoinThreshold in spot-ml
+ 
+ After Spark LDA runs, Topics Matrix and Topics Distribution are joined with the original data set  i.e. NetFlow records, DNS records or Proxy records to determine the probability of each event to happen. This joining  process is similar to join a big data set and a lookup table. In this case, the big data set is the entire set of records, and the lookup table is a dictionary of documents and probabilities per topic or words and probabilities per topic. 
+ 
+ Because of the possible diversity of documents/IPs, the lookup table containing document probability distribution can grow to something bigger than 10 MB. Taking in account that 10 MB is Spark's default auto broadcast threshold for joins, a join with a lookup table bigger than that threshold will result in the execution of a traditional join with lots of shuffling. 
+ 
+The correct setting of  SPK_AUTO_BRDCST_JOIN_THR and PRECISION can help to always broadcast document probability distribution lookup table and avoid slow joins. 
+
+As a first step, users need to decide whether they want to change from 64 bit floating point probabilities to 32 bit floating point probabilities; if users decide to change from 64 to 32 bit, the document probability distribution lookup table will be half the size and more easily broadcasted. 
+
+If users want to cut the payload on half, they should set precision option to 32.
```

Comment:
If users want to cut payload memory consumption roughly in half, they should set the precision option to 32